src/error.rs

```diff
@@ -483,6 +490,11 @@ pub enum ErrorKind {
     /// No resume token was present in a change stream document.
     #[error("Cannot provide resume functionality when the resume token is missing")]
     MissingResumeToken,
+
+    /// An error occurred during encryption or decryption.
+    #[cfg(feature = "fle")]
+    #[error("An error occurred during client-side encryption: {0}")]
+    Crypt(mongocrypt::error::Error),
```

Comment:
I think it might be a bit clearer if this errorkind `Fle` or `QueryableEncryption`, depending on what we call things these days. Alternatively, if this is to encompass only errors from `mongocrypt` and not more general, driver-side encryption errors, maybe we name this one `Mongocrypt` and have another kind for regular fle ones? Also happy to defer this question until further in the implementation when we know more about the kinds of errors we'll be returning.

Reply:
Renamed this to `Csfle` to match elsewhere.